FAQ.md

```diff
@@ -0,0 +1,163 @@
+[`PhoneNumberUtil`]
+(https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java).
+
+### Why does PhoneNumberUtil return false for valid short numbers?
+
+Short numbers are out of scope of
+[`PhoneNumberUtil`]
+(https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java).
+For short numbers, use
+[`ShortNumberInfo`]
+(https://github.com/googlei18n/libphonenumber/blob/master/java/libphonenumber/src/com/google/i18n/phonenumbers/ShortNumberInfo.java).
+
+### What does it mean for a phone number to be valid?
+
+Our phone number library can tell that a number range is valid when there is
+sufficient official documentation, with some latency after this fact is brought
+to our attention via issue reports or notifications (see below for more
+information on where our metadata comes from). A valid number range is one from
+which numbers can be freely assigned by carriers to users.
+
+Do not rely on libphonenumber to determine whether numbers are currently
+assigned to a specific user and reachable. Some products (e.g.
+[Google 2-step verification] (https://www.google.com/landing/2step/)) do
+this with a verification step e.g. by sending an SMS or placing an automated
+phone call with a verification code). This is not technically feasible without
+such a verification step given the complicated international world we live in,
+with varying standardization practices in different regions.
+
+### When should I use isValidNumberForRegion?
+
+Rarely! Many people have phone numbers that do not belong to the country they
+live in. This applies to particularly to mobile numbers, but may also be true
+for VoIP numbers etc. Note also that the regions your application supports may
+not be the same as the regions we support. For example, the Channel Islands such
+as "Jersey" have their own region code - JE. If you allow these users to sign up
+as a British user ("GB"), their phone numbers will not be considered valid for
+the region "JE".
+
+One use-case where this method may be useful is if you want to see if a
+`FIXED_LINE` number for a business matches the country it is in, to try and spot
+data errors.
+
+### What types of phone numbers can SMSs be sent to?
+
+SMSs can be sent to `TYPE_MOBILE` or `TYPE_FIXED_LINE_OR_MOBILE` numbers. However,
+in some countries it is possible to configure other types, such as normal
+land-lines, to receive SMSs.
+
+### Why did I get `TYPE_FIXED_LINE_OR_MOBILE` as the type of my phone number?
+
+Some number ranges are explicitly defined as being for fixed-line or mobile
+phones. We even represent ranges defined as being "Mostly land-line" in this
+way.
+
+### What about M2M (machine to machine) numbers?
+
+libphonenumber does not support M2M numbers at the moment, but might in the
+future.
+
+One of the reasons libphonenumber doesn't supported M2M so far is because no one
```

Comment:
Done

On Wed, Mar 2, 2016 at 4:34 PM, Keghani Kouzoujian <notifications@github.com
> wrote:

> In FAQ.md
> <https://github.com/googlei18n/libphonenumber/pull/987#discussion_r54739072>
> :
>
> > +SMSs can be sent to `TYPE_MOBILE` or `TYPE_FIXED_LINE_OR_MOBILE` numbers. However,
> > +in some countries it is possible to configure other types, such as normal
> > +land-lines, to receive SMSs.
> > +
> > +### Why did I get `TYPE_FIXED_LINE_OR_MOBILE` as the type of my phone number?
> > +
> > +Some number ranges are explicitly defined as being for fixed-line or mobile
> > +phones. We even represent ranges defined as being "Mostly land-line" in this
> > +way.
> > +
> > +### What about M2M (machine to machine) numbers?
> > +
> > +libphonenumber does not support M2M numbers at the moment, but might in the
> > +future.
> > +
> > +One of the reasons libphonenumber doesn't supported M2M so far is because no one
>
> s/doesn't/hasn't
>
> or
> s/supported/support
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/libphonenumber/pull/987/files#r54739072>.
>